rclcpp/src/rclcpp/node_interfaces/node_graph.cpp

```diff
@@ -178,6 +179,136 @@ NodeGraph::get_service_names_and_types_by_node(
   return services_and_types;
 }
 
+std::map<std::string, std::vector<std::string>>
+NodeGraph::get_client_names_and_types_by_node(
+  const std::string & node_name,
+  const std::string & namespace_) const
+{
+  rcl_names_and_types_t service_names_and_types = rcl_get_zero_initialized_names_and_types();
+  rcl_allocator_t allocator = rcl_get_default_allocator();
```

Comment:
It would be good if an allocator could be provided somehow.
Though it seems a bit unrelated to the PR.

Reply:
probably, we can have `allocator` in `NodeGraph` class. let's skip this one since it is not really related to the fix.